### PR TITLE
Merge CSS classes in the "root page dependent modules" module

### LINKED ISSUE #5535: Insert module "root page dependent modules" via content element does not merge css classes
### Affected version(s)

4.13

### Description

When the Root Page Dependent Module is inserted via a content element, css classes are not merged.

### Reproducible by creating the following modules
- Module A: css class `module-a`
- Module B: css class `module-b`
- Module C (Root page dependent modules): with A and B
- Content Element (Module): Inserts module C and assigns css class `test`

Only the classes `module-a` or `module-b` are output, `test` is ignored

Is this the expected behavior? Then we should consider removing the ID/CSS class selection for this content element.

### PR DESCRIPTION
**Pull request by bytehead**
Fixes #5535

### CODE REVIEW

File: core-bundle/src/Controller/FrontendModule/RootPageDependentModulesController.php [hunk 1]
```diff
@@ -37,8 +37,15 @@ public function __invoke(Request $request, ModuleModel $model, string $section,
             return new Response('');
         }
 
+        $moduleModel = $this->container->get('contao.framework')->getAdapter(ModuleModel::class);
+        $module = $moduleModel->findByPk($modules[$pageModel->rootId]);
+
+        $cssID = StringUtil::deserialize($module->cssID, true);
```
Comment (ausi) [inline core-bundle/src/Controller/FrontendModule/RootPageDependentModulesController.php:46]: The ID should get overwritten if set I think.

```suggestion
        $cssID = StringUtil::deserialize($module->cssID, true);
        if ($idAttribute = $request->attributes->get('templateProperties', [])['cssID'] ?? null) {
            $cssID[0] = substr($idAttribute, 5, -1);
        }
```

File: core-bundle/src/Controller/FrontendModule/RootPageDependentModulesController.php [hunk 2]
```diff
@@ -37,8 +37,15 @@ public function __invoke(Request $request, ModuleModel $model, string $section,
...
+        $cssID[1] = trim(sprintf('%s %s', $cssID[1], implode(' ', $model->classes)));
```
Comment (ausi) [inline core-bundle/src/Controller/FrontendModule/RootPageDependentModulesController.php]: Possibly undefined array key.

```suggestion
        $cssID[1] = trim(sprintf('%s %s', $cssID[1] ?? '', implode(' ', $model->classes)));
```

File: core-bundle/src/Controller/FrontendModule/RootPageDependentModulesController.php [hunk 3]
```diff
@@ -37,8 +37,15 @@ public function __invoke(Request $request, ModuleModel $model, string $section,
...
+        $module->cssID = $cssID;
```
Comment (ausi) [inline core-bundle/src/Controller/FrontendModule/RootPageDependentModulesController.php:54]: `cssID` in the model is usually serialized AFAIK.

```suggestion
        $module->cssID = serialize($cssID);
```

Comment (bytehead) [reply core-bundle/src/Controller/FrontendModule/RootPageDependentModulesController.php:54]: It is, but not needed, see https://github.com/contao/contao/blob/4.13/core-bundle/src/Resources/contao/elements/ContentModule.php#L78.

I tried both, makes no difference 😅

File: core-bundle/src/Controller/FrontendModule/RootPageDependentModulesController.php (hunk 1 above)
Comment (bytehead) [reply core-bundle/src/Controller/FrontendModule/RootPageDependentModulesController.php:46]: Adjusted in 35fa17cb4f.

File: core-bundle/src/Controller/FrontendModule/RootPageDependentModulesController.php (hunk 2 above)
Comment (bytehead) [reply core-bundle/src/Controller/FrontendModule/RootPageDependentModulesController.php]: Fixed in 8d4482a4a5.

**Review by ausi — DISMISSED** [review]

**Review by leofeyer — APPROVED** [review]

**Comment by leofeyer** [comment]
Thank you @bytehead.

**Pull request merged**